File: CleanAddisBackOfficeApi/src/main/java/com/iyoa/cleanaddis/repository/ReportRepository.java
package com.iyoa.cleanaddis.repository;

import java.util.List;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import com.iyoa.cleanaddis.model.Article;
import com.iyoa.cleanaddis.model.Report;

public interface ReportRepository extends JpaRepository<Report, UUID> {
	@Query(value = "SELECT * FROM report "
			+ "WHERE username =:username "
			, nativeQuery = true)
	List<Report> findReportByUsername(String username);

}

File: app/src/main/java/com/iyoa/cleanaddis/repository/report/ReportRepository.kt
package com.iyoa.cleanaddis.repository.report

import androidx.lifecycle.LiveData
import com.iyoa.cleanaddis.data.report.Report
import com.iyoa.cleanaddis.data.report.ReportDAO

class ReportRepository( private val reportDAO: ReportDAO) {

    fun getReportByUuid(uuid:Long): LiveData<Report> = reportDAO.getReportByUuid(uuid)


    fun getAllReports(username:String): LiveData<List<Report>> = reportDAO.getAllReports(username)

    fun getAllReport(): LiveData<List<Report>> = reportDAO.getAllReport()


    fun addReport(report:Report) = reportDAO.addReport(report)

    fun editReport(report:Report) = reportDAO.updateReport(report)

    fun deleteReport(report:Report) = reportDAO.deleteReport(report)

    fun addReports(report: List<Report>)  = reportDAO.addReports(report)


}